app/models/manageiq/providers/azure/event_catcher_mixin.rb

```diff
@@ -0,0 +1,38 @@
+module ManageIQ::Providers::Azure::EventCatcherMixin
+  def parse_event_type(event)
+    event_name = event["eventName"]["value"]
+    event_type = ""
+
+    unless event["authorization"].nil? || event["authorization"]["action"].nil?
```

Comment:
I think you can skip these `nil` checks and then have `parse_event_action` handle it (see below).